tools/gritsettings/filter_resource_ids.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import argparse
+import re
+import sys
+
+filter_values = ['glic', 'ash', 'chromeos']
+start_pattern = [r'^{}\/'.format(val) for val in filter_values]
+other_pattern = [r'\/{}\/'.format(val) for val in filter_values]
+patterns = '|'.join(start_pattern + other_pattern)
+
+exclusions = [
+    # This GRD is built on all Desktop platforms for some reason
+    # From upstream comment:
+    # TODO(b/207518736): Input overlay resources will be changed
+    # to proto soon, thus not going to move this resource pak to
+    #  under ash.
+    'chromeos/ash/experiences/arc/input_overlay/resources/input_overlay_resources.grd'
+]
+
+
+def read_file(filename):
```

Comment:
can we add some explanation of what this script does? (the whole thing, not just read_file)

Reply:
@mkarolin Mind adding that?

Reply:
Added a comment in https://github.com/brave/brave-core/pull/32111/commits/f87a955a703cdb6913bc6b1be5455a176cdc6d83